src/Resources/ResourceManager/Utilities/BicepUtility.cs

```diff
@@ -0,0 +1,63 @@
+﻿
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Exceptions;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Utilities
+{
+    public static class BicepUtility
+    {
+        public static bool IsBicepExecutable = false;
+
+        public static bool IsBicepFile(string templateFilePath)
+        {
+            return ".bicep".Equals(Path.GetExtension(templateFilePath)?.ToLower());
+        }
+
+        public delegate List<T> ScriptExecutor<T>(string script);
+
+        public static bool CheckBicepExecutable<T>(ScriptExecutor<T> executeScript)
+        {
+            try
+            {
+                executeScript("get-command bicep");
+            }
+            catch
+            {
+                IsBicepExecutable = false;
+                return IsBicepExecutable;
+            }
+            IsBicepExecutable = true;
+            return IsBicepExecutable;
+        }
+
+        public static string BuildFile<T>(ScriptExecutor<T> executeScript, string bicepTemplateFilePath)
+        {
+            if (!IsBicepExecutable && !CheckBicepExecutable(executeScript))
+            {
+                throw new AzPSApplicationException(Properties.Resources.NoBicepInstalled);
+            }
+
+            executeScript($"bicep build {bicepTemplateFilePath}");
+
+            string jsonTemplateFilePath = Path.Combine(Path.GetDirectoryName(bicepTemplateFilePath),
+                Path.GetFileNameWithoutExtension(bicepTemplateFilePath) + ".json");
+
+            return jsonTemplateFilePath;
+        }
+
+        public static string BuildFileContent<T>(ScriptExecutor<T> executeScript, string templateContent, out string jsonTemplateFilePath)
+        {
+            string tempFilePath = Path.GetTempFileName();
+            FileUtilities.DataStore.WriteFile(tempFilePath, templateContent);
+            jsonTemplateFilePath = BuildFile(executeScript, tempFilePath);
+            return FileUtilities.DataStore.ReadFileAsText(jsonTemplateFilePath);
```

Comment:
Should we delete jsonTemplateFilePath before returning?

Reply:
We shouldn't, this jsonFile will be used in the future.